be/src/pipeline/exec/hashjoin_build_sink.cpp

```diff
@@ -222,24 +231,25 @@ Status HashJoinBuildSinkLocalState::close(RuntimeState* state, Status exec_statu
         }
     }};
 
-    if (!_runtime_filter_producer_helper || state->is_cancelled() || !_eos) {
-        return Base::close(state, exec_status);
-    }
-
     try {
-        RETURN_IF_ERROR(_runtime_filter_producer_helper->process(
-                state, _shared_state->build_block.get(), p._use_shared_hash_table,
-                p._runtime_filters));
+        if (!_terminated && _runtime_filter_producer_helper && !state->is_cancelled()) {
```

Comment:
not keep origin `eos` check ? 
1. what situation the `eos == false` we call close?
2. if not eos, should we build hash table